doc/hardware_acceleration.rst

```diff
@@ -0,0 +1,157 @@
+Hardware Acceleration
+==============================
+
+Botan provides built-in support for hardware acceleration of certain algorithms
+on certain platforms. These alternate implementations use special CPU instructions
+that are not available on all platforms and either speed up the algorithm
+or improve security in terms of side channel resistance.
+
+A “base” software implementation is always provided. For example, for the AES-128
+block cipher three implementations are available. All of them are immune to
+common cache/timing based side channels.
```

Comment:
This is slightly confusing in that someone could misread it as "All of [the "base" software implementations of all ciphers]" vs "All of [the AES implementations]"

Reply:
Made it more clear that it only applies to the AES-128 example.